sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeFixedWidthAggregationMapSuite.scala

```diff
@@ -140,7 +159,21 @@ class UnsafeFixedWidthAggregationMapSuite
     map.free()
   }
 
-  test("test sorting") {
+  testWithMemoryLeakDetection("test external sorting") {
+    // Calling this make sure we have block manager and everything else setup.
+    TestSQLContext
+
+    TaskContext.setTaskContext(new TaskContextImpl(
```

Comment:
Should we use a `finally` block to ensure that the TaskContext is cleared at the end of this test?